Program.cs

```diff
@@ -53,95 +70,280 @@ static void Main(string[] args)
+        }
+
+        private void CleanStore(
+            string partitionSnapshotPath,
+            string stateSnapshotPath,
+            string storePath,
+            string partitionDirectory,
+            string stateDirectory)
+        {
+            if (File.Exists(partitionSnapshotPath))
             {
-                throw new CommandExitedException("Canonical chain doesn't exists.", -1);
+                File.Delete(partitionSnapshotPath);
+            }
+
+            if (File.Exists(stateSnapshotPath))
+            {
+                File.Delete(stateSnapshotPath);
+            }
+
+            var blockPerceptPath = Path.Combine(storePath, "blockpercept");
+            if (Directory.Exists(blockPerceptPath))
+            {
+                Directory.Delete(blockPerceptPath, true);
+            }
+
+            var stagedTxPath = Path.Combine(storePath, "stagedtx");
+            if (Directory.Exists(stagedTxPath))
+            {
+                Directory.Delete(stagedTxPath, true);
+            }
+
+            if (Directory.Exists(partitionDirectory))
+            {
+                Directory.Delete(partitionDirectory, true);
+            }
+
+            if (Directory.Exists(stateDirectory))
+            {
+                Directory.Delete(stateDirectory, true);
+            }
+        }
+
+        private void CleanPartitionStore(string partitionDirectory)
+        {
+            var statePath = Path.Combine(partitionDirectory, "state");
+            if (Directory.Exists(statePath))
+            {
+            Directory.Delete(statePath, true);
+            }
+
+            var stateHashesPath = Path.Combine(partitionDirectory, "state_hashes");
+            if (Directory.Exists(stateHashesPath))
+            {
+            Directory.Delete(stateHashesPath, true);
+            }
+
+            var stateRefPath = Path.Combine(partitionDirectory, "stateref");
+            if (Directory.Exists(stateRefPath))
+            {
+            Directory.Delete(stateRefPath, true);
+            }
+
+            var statesPath = Path.Combine(partitionDirectory, "states");
+            if (Directory.Exists(statesPath))
+            {
+            Directory.Delete(statesPath, true);
+            }
+
+            var chainPath = Path.Combine(partitionDirectory, "chain");
+            if (Directory.Exists(chainPath))
+            {
+            Directory.Delete(chainPath, true);
+            }
+
+            var blockPath = Path.Combine(partitionDirectory, "block");
+            var blockIndexPath = Path.Combine(blockPath, "blockindex");
+            if (Directory.Exists(blockIndexPath))
+            {
+            Directory.Delete(blockIndexPath, true);
+            }
+
+            var txPath = Path.Combine(partitionDirectory, "tx");
+            var txIndexPath = Path.Combine(txPath, "txindex");
+            if (Directory.Exists(txIndexPath))
+            {
+            Directory.Delete(txIndexPath, true);
+            }
```

Comment:
1. How about indenting one tab (i.e. 2 space or 4 space) the code inner `if` statement?
1. Also how about removing duplicated codes with `foreach` statement?

As code...

```suggestion
            var cleanDirectories = new[]
            {
                Path.Combine(partitionDirectory, "state"),
                Path.Combine(partitionDirectory, "state_hashes"),
                Path.Combine(partitionDirectory, "stateref"),
                Path.Combine(partitionDirectory, "states"),
                Path.Combine(partitionDirectory, "chain"),
                Path.Combine(partitionDirectory, "block", "blockindex"),
                Path.Combine(partitionDirectory, "tx", "txindex"),
            };

            foreach (var path in cleanDirectories)
            {
                if (Directory.Exists(path))
                {
                    Directory.Delete(path, true);
                }
            }
```